src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs

```diff
@@ -296,6 +304,91 @@ void AddSection(string kind, ImmutableArray<TaggedText> taggedParts)
             return default;
         }
 
+        private static ImmutableArray<TaggedText> GetConstantValueContent(
+            SyntaxToken token,
+            SemanticModel semanticModel,
+            ISyntaxFactsService syntaxFacts,
+            ISymbolDisplayService displayService,
+            CancellationToken cancellationToken)
+        {
+            if (!syntaxFacts.IsBinaryExpression(token.Parent))
+            {
+                return default;
+            }
+
+            syntaxFacts.GetPartsOfBinaryExpression(token.Parent, out var leftExpression, out var operatorToken, out var rightExpression);
+
+            if (token != operatorToken)
+            {
+                return default;
+            }
+
+            var constant = semanticModel.GetConstantValue(token.Parent, cancellationToken);
+            if (!constant.HasValue)
+            {
+                return default;
+            }
+
+            var leftConstant = semanticModel.GetConstantValue(leftExpression, cancellationToken);
+            if (!leftConstant.HasValue)
+            {
+                return default;
+            }
+
+            var rightConstant = semanticModel.GetConstantValue(rightExpression, cancellationToken);
+            if (!rightConstant.HasValue)
+            {
+                return default;
+            }
+
+            var textBuilder = ImmutableArray.CreateBuilder<TaggedText>();
+            textBuilder.AddLineBreak();
+            textBuilder.AddText(FeaturesResources.Constant_value_colon);
+            textBuilder.AddSpace();
+            textBuilder.AddRange(FormatValue(leftConstant.Value));
+            textBuilder.AddSpace();
+            textBuilder.Add(
+                new TaggedText(
+                    syntaxFacts.IsKeyword(operatorToken) ? TextTags.Keyword : TextTags.Operator,
+                    operatorToken.Text));
+            textBuilder.AddSpace();
+            textBuilder.AddRange(FormatValue(rightConstant.Value));
+            textBuilder.AddSpace();
+            textBuilder.AddOperator("=");
+            textBuilder.AddSpace();
+            textBuilder.AddRange(FormatValue(constant.Value));
+            return textBuilder.ToImmutable();
+
+            ImmutableArray<TaggedText> FormatValue(object value)
+            {
+                var taggedText = displayService.FormatPrimitiveToDisplayParts(value, options: default).ToTaggedText();
+                return TrimTaggedTextRun(taggedText, 42);
+            }
+        }
+
+        private static ImmutableArray<TaggedText> TrimTaggedTextRun(ImmutableArray<TaggedText> taggedTextRun, int maxLength)
+        {
+            const string UnicodeEllipsis = "\u2026";
```

Comment:
feels like it would be better to me if we just showed the final constant value.

Reply:
Ask @sharwell, he's been silent about that (and it seems it wasn't brought up in the design review).

Reply:
Putting on my IDE designer hat: there is value in showing the constant value.  Let's *start* with that and see if there's really a compelling need to do more.  Absent that, we should just go with something simpler.

Reply:
You do not need to convince me - I agree that would be preferrable here. You need to convince @sharwell.